File: api/src/main/java/dev/abarnes/api/model/BlogPost.java
package dev.abarnes.api.model;

import java.time.Instant;
import java.util.List;

public record BlogPost(
    String id,
    String title,
    String slug,
    String content,
    String summary,
    List<String> tags,
    Instant createdAt,
    Instant updatedAt
) {}

File: web/src/types/BlogPost.ts
export interface BlogPost {
  id: string;
  title: string;
  slug: string;
  content: string;
  summary: string;
  tags: string[];
  createdAt: string;
  updatedAt: string;
}

export interface BlogPostCreate {
  title: string;
  content: string;
  summary: string;
  tags: string[];
}

export interface BlogPostUpdate extends Partial<BlogPostCreate> {
  id: string;
}
